### PR TITLE
fix: vulnerabilities in dependencies

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

https://linear.app/rudderstack/issue/SDK-2575/address-vulnerabilities-in-node-sdk

I've updated all the dependencies to the latest.

Some notable changes:
- Upgrade Node to the latest version.
- Removed unwanted Babel dependencies.
- Formatted and fixed lint issues in the code.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -86,6 +86,11 @@ export type TrackParams = {
   timestamp?: Date;
 } & IdentityOptions;
 
+export type TrackMessage<PropertiesType> = Omit<TrackParams, 'event' | 'properties'> & {
+  event?: string;
+  properties: PropertiesType;
+} & Record<string, any>;
+
```
Comment (MoumitaM) [inline index.d.ts:93]: Here we are making the event as optional parameter in track call. What is the motivation for this change?

**Review by MoumitaM — CHANGES_REQUESTED** [review]
Have we tested if the SDK is building properly with the new changes?

File: index.d.ts (hunk 1 above)
Comment (saikumarrs) [reply index.d.ts]: I'm surprised this change was made. I've removed the whole type as it is not needed.

**Review by MoumitaM — APPROVED** [review]
